### DISCUSSION TITLE
Which version of Microsoft.WindowsAppSDK?

### DISCUSSION THREAD

**Discussion by Orgbrat** [dotnet/maui#2921] [Q&A]
Updated (11/08/2021):

Have a solution with a MAUI single project preview 9 application created in Visual Studio 2022 Preview 7.0 and it uses WindowsAppSDK 1.0.0-experimental1. I have upgraded to Visual Studio 2022 17.1.0 Preview 1.0 which updates MAUI to Preview 10. The  MAUI application fails to build now with the error below.

Error	NU1605	Detected package downgrade: Microsoft.WindowsAppSDK from 1.0.0-preview3 to 1.0.0-experimental1. Reference the package directly from the project to select a different version. 

How do I update my MAUI solution to correctly use Microsoft.WindowsAppSDK 1.0.0-preview3 correctly now?

Update:

Created a new Visual Studio 2022 17.1.0 Preview 1.0 MAUI application (default template) and it gives the same error as above...

Orgbrat

**Comment by DaveNatalie** [comment]
Any luck finding a workaround for this?

**Reply by Orgbrat** [reply]
Yep, follow the [Preview 9 to Preview 10 Migration guide](https://github.com/dotnet/maui/wiki/Migrating-from-Preview-9-to-10).
It moves the WindowsAppSDK to 1.0.0-preview3...

**Reply by DaveNatalie** [reply]
I created an app from scratch using the templates in Visual Studio. It looks like the templates are out of date.

I ended up fixing the issue by installing a third party template plugin.

https://marketplace.visualstudio.com/items?itemName=egvijayanand.maui-templates

**Comment by VladislavAntonyuk** [comment]
You can even use stable version. Preview 10 works with it.
You can create new Maui app and copy csproj code to your app.

**Reply by DaveNatalie** [reply]
I got this error when creating a new Maui app using the template in Visual Studio. Does it only work from the command line? Is there a way to update my maui templates to the latest version (I am using the latest version of Visual Studio 2022 Preview 1.1).

**Reply by VladislavAntonyuk** [reply]
run `maui-check`. looks like you have outdated templates
